### ISSUE TITLE
[Proposal][BREAKING] Global Stringify

### ISSUE THREAD

**Issue by felangel** [felangel/equatable#68]
**Is your feature request related to a problem? Please describe.**
It would be nice to have a way to globally configure `stringify` for all `Equatable` instances. The proposal includes a breaking change to refactor `stringify` from a `bool` to an `enum` in order to be able to support a "default" (normal) setting in addition to an "enabled" (always) and "disabled" (never) setting.

**Describe the solution you'd like**

```dart
enum Stringify { normal, always, never }

// override stringify at the individual Equatable instance
class Credentials extends Equatable {
  final String username;
  final String password;

  const Credentials({this.username, this.password});

  @override
  List<Object> get props => [username, password];

  @override
  Stringify get stringify => Stringify.always;
}

// override stringify for all Equatable instances
EquatableConfig.stringify = Stringify.always;
```

If `stringify` is overridden for an individual `Equatable` it will take precedence over the `EquatableConfig` value.

Addresses #60 and #65

**Comment by zepfietje** [comment]
@felangel could you elaborate on the behavior of the "default" (normal) setting?

**Comment by felangel** [comment]
@zepfietje the normal setting will result in just the runtimeType being printed.

**Comment by zepfietje** [comment]
@felangel what's the difference between `normal` and `never`?

**Comment by felangel** [comment]
Normal will defer to the EquatableConfig setting never will never stringify.

**Comment by zepfietje** [comment]
So why exactly would we need an `enum`? Isn't there a way to defer to the `EquatableConfig` by default and allow the `stringify` `bool` to be overridden for an individual `Equatable`?

**Comment by felangel** [comment]
@zepfietje good call...I've updated the issue to maintain a bool (which makes this a non-breaking change). The PR is open at #69 👍

**Comment by zepfietje** [comment]
@felangel this issue can now be unpinned. 👍
